process-migration-service/src/main/java/org/kie/processmigration/rest/PlanResource.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.processmigration.rest;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import org.kie.processmigration.model.Plan;
+import org.kie.processmigration.service.PlanService;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+
+@Path("/plans")
+@Api(value = "Plans")
+@Produces(MediaType.APPLICATION_JSON)
+@ApplicationScoped
+public class PlanResource {
+
+    @Inject
+    private PlanService planService;
+
+    @GET
+    @ApiOperation(value = "Get all existing Migration plans")
+    public Response findAll() {
+        return Response.ok(planService.findAll()).build();
+    }
+
+    @GET
+    @Path("/{id}")
+    @ApiOperation(value = "Finds a migration plan by the given plan Id")
+    public Response get(@ApiParam(value = "Plan Id") @PathParam("id") Long id) {
+        Plan plan = planService.get(id);
+        if (plan == null) {
+            return getPlanNotFound(id);
```

Comment:
similar as for migration service ... throw the not found exception from within the service

Reply:
Same. It's not an exception but a Response object.